build/Build.cs

```diff
@@ -139,9 +142,19 @@ partial class Build : NukeBuild
       .DependsOn(Tests)
       .Executes(() =>
       {
-          //Since this is about a new release, `RunChangeLog` need to be executed to update the ChangeLog.md with the new version
-          //from which LatestVersion will be parsed
           var version = GitVersion.SemVer;
+          var branchName = GitVersion.BranchName;
+
+          if(branchName.Equals(ReleaseBranch, StringComparison.OrdinalIgnoreCase) 
+          && !GitVersion.MajorMinorPatch.Equals(LatestVersion.Version.ToString()))
+          {
+              // Force CHANGELOG.md in case it skipped the mind
+              Assert.Fail($"CHANGELOG.md needs to be update for final release. Current version: '{LatestVersion.Version}'. Next version: {GitVersion.MajorMinorPatch}");
+          }
+          var releaseNotes = branchName.Equals(ReleaseBranch, StringComparison.OrdinalIgnoreCase) 
```

Comment:
If we are not in the ReleaseBranch, let's make use of the ReleaseNote in `Directory.Build.props` since this is a pre-release.